src/client.ts

```diff
@@ -263,6 +265,34 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     });
   }
 
+  getUserContextInstance(userInfo: UserInfo): optimizely.OptimizelyUserContext | null {
+    let userContext: optimizely.OptimizelyUserContext | null = null;
+
+    const userId = userInfo.id || this.user.id;
+    const userAttributes = userInfo.attributes || this.user.attributes;
+
+    if (this.userContext) {
+      if (JSON.stringify(userInfo) === JSON.stringify(this.user)) {
```

Comment:
using `JSON.stringify` to compare objects can cause undesired results. I would prefer explicitly coparing id and attributes. You can create a separate comparison method to make the code clearner.